ts/features/itwallet/issuance/hooks/useEidEventsTracking.ts

```diff
@@ -54,7 +55,23 @@ export const useEidEventsTracking = ({ failure, identification }: Params) => {
     }
 
     if (failure.type === IssuanceFailureType.UNEXPECTED) {
-      return trackItwIdRequestUnexpectedFailure(failure);
+      /*
+       * We handle two cases here:
+       * 1. If failure.reason is undefined/null, we serialize the failure to provide a default message with "Reason not provided".
+       * 2. If failure.reason is an empty object with no keys, we serialize it to extract message property.
+       * To maintain compatibility with the existing failure tracking, we keep the original `failure` object when `failure.reason` is not empty.
+       */
+      if (!failure.reason) {
+        return trackItwIdRequestUnexpectedFailure(
+          serializeFailureReason(failure)
+        );
+      }
+      return trackItwIdRequestUnexpectedFailure(
+        typeof failure.reason === "object" &&
+          Object.keys(failure.reason).length === 0
+          ? serializeFailureReason(failure)
+          : failure
+      );
```

Comment:
I'd try to be more descriptive with names. The check `typeof failure.reason === "object" && Object.keys(failure.reason).length === 0` might not be straightforward, especially when mixed with the other code.
```suggestion
const isNonDefaultSerializable =
    typeof failure.reason === "object" && Object.keys(failure.reason).length === 0

const shouldSerializeReason = !failure.reason || isNonDefaultSerializable

return trackItwIdRequestUnexpectedFailure(
    shouldSerializeReason ? serializeFailureReason(failure) : failure
);
```